lib/src/verifiers/nodes/test-verifier.spec.ts

```diff
@@ -0,0 +1,397 @@
+  const endpointNode: EndpointNode = {
+    name: fakeLocatable("SomeEndpoint"),
+    method: fakeLocatable<HttpMethod>("POST"),
+    path: fakeLocatable("/company/:companyId/users"),
+    request: fakeLocatable({
+      headers: fakeLocatable([
+        fakeLocatable<HeaderNode>({
+          name: fakeLocatable("x-region"),
+          type: {
+            kind: TypeKind.STRING
+          },
+          optional: false
+        })
+      ]),
+      pathParams: fakeLocatable([
+        fakeLocatable<PathParamNode>({
+          name: fakeLocatable("companyId"),
+          type: {
+            kind: TypeKind.STRING
+          }
+        })
+      ]),
+      body: fakeLocatable<BodyNode>({
+        type: {
+          kind: TypeKind.OBJECT,
+          properties: [
+            {
+              name: "name",
+              type: {
+                kind: TypeKind.STRING
+              },
+              optional: false
+            }
+          ]
+        }
+      })
+    }),
+    responses: [
+      fakeLocatable({
+        status: fakeLocatable(201)
+      }),
+      fakeLocatable({
+        status: fakeLocatable(400)
+      })
+    ],
+    tests: []
+  };
+
+  test("valid for correct usage", () => {
+    const testNode = fakeLocatable({
+      request: fakeLocatable<TestRequestNode>({
+        headers: [
+          {
+            name: "x-region",
+            expression: { kind: TypeKind.STRING_LITERAL, value: "AU" }
+          }
+        ],
+        pathParams: [
+          {
+            name: "companyId",
+            expression: { kind: TypeKind.STRING_LITERAL, value: "cave" }
+          }
+        ],
+        body: {
+          kind: TypeKind.OBJECT,
+          properties: [
+            {
+              name: "name",
+              expression: { kind: TypeKind.STRING_LITERAL, value: "Spider" }
+            }
+          ]
+        }
+      }),
+      response: fakeLocatable({
+        status: fakeLocatable(201)
+      }),
+      options: {
+        allowInvalidRequest: false
+      }
+    });
+    expect(verifyTestNode(testNode, endpointNode, [])).toHaveLength(0);
+  });
+
+  test("valid when invalid request is allowed", () => {
+    const testNode = fakeLocatable({
+      request: fakeLocatable<TestRequestNode>({
+        headers: [],
+        pathParams: []
+      }),
+      response: fakeLocatable({
+        status: fakeLocatable(201)
+      }),
+      options: {
+        allowInvalidRequest: true
+      }
+    });
+    expect(verifyTestNode(testNode, endpointNode, [])).toHaveLength(0);
+  });
+
+  test("invalid with no matching response", () => {
+    const testNode = fakeLocatable({
+      request: fakeLocatable<TestRequestNode>({
+        headers: [],
+        pathParams: []
+      }),
+      response: fakeLocatable({
+        status: fakeLocatable(204)
+      }),
+      options: {
+        allowInvalidRequest: true
+      }
+    });
+    expect(verifyTestNode(testNode, endpointNode, [])).toHaveLength(1);
```

Comment:
Why not expect a specific error message? (same below)

My reasoning is that it's good to check that an error is returned, but even better to check that the error is what we expect it to be.